### PR TITLE
Re-implements the XLSX/CSV uploader

### LINKED ISSUE #327: Dataset API broken
**Describe the bug**
The Dataset API is broken due to an unexpected item in `UserUtils.proccess_create_or_modify_user_request` function. Should fix this and add basic functional tests. I believe removing `force_dict_keys_lowercase=True` will fix this API.

https://github.com/teamsempo/SempoBlockchain/blob/efeef9c5753e75aac0e6cfcac1a63995d4c9d69e/app/server/api/dataset_api.py#L83-L85

**To Reproduce**
Steps to reproduce the behaviour:
1. Upload a spreadsheet
2. Follow steps and submit
4. Returns 500 error

**Additional context**
Review API and deprecate if not needed.

### PR DESCRIPTION
**Pull request by tristanhcole**
**Describe the Pull Request**
Screen Shot 2020-12-08 at 5 29 54 pm

Re-implements the XLSX/CSV uploader and adds functional tests. Improves import UI by using ant design.
Closes #327 .  TODO: add to docs.withsempo.com

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by tristanhcole** [comment]
[Excel Import](https://trello.com/c/BJmLqq0q/715-excel-import)

File: app/client/components/pages/uploadPage.jsx [hunk 1]
```diff
@@ -7,69 +7,29 @@ import UploadedTable from "../uploader/uploadedTable.jsx";
...
-
-const uploadPage = ({ loggedIn, spreadsheetData, location }) => {
+const uploadPage = ({ spreadsheetData, location }) => {
   const is_vendor = location.search.indexOf("type=vendor") !== -1;
 
   if (spreadsheetData) {
-    var inner_div = (
-      <UploadedTable data={spreadsheetData} is_vendor={is_vendor} />
-    );
+    return <UploadedTable data={spreadsheetData} is_vendor={is_vendor} />;
```
Comment (enjeyw) [inline app/client/components/pages/uploadPage.jsx:18]: don't approve unless is_vendor works (or rather, the user can specify the user type)

Comment (michielderoos) [reply app/client/components/pages/uploadPage.jsx:18]: @tristanhcole Is this reviewable yet?

Comment (enjeyw) [reply app/client/components/pages/uploadPage.jsx:18]: got vendor specification working by just allowing an `account_type` column

**Review by enjeyw — APPROVED** [review]
added csv upload compatability because it didn't actually exist lol. was only like 3 lines

**Pull request merged**